### PR TITLE
Fix for NVHPC

### PR DESCRIPTION
**Pull request by benjaminmenetrier**
## Description

The call to `register_variable_attribute` is not compiling with `nvhpc-22.5`. This PR adds a preprocessor condition to compile anyway and get a clean abort if this routine is called.

### CODE REVIEW

File: fms2_io/netcdf_io.F90 [hunk 1]
```diff
@@ -2143,7 +2143,7 @@ subroutine write_restart_bc(fileobj, unlim_dim_level)
   class(FmsNetcdfFile_t), intent(inout) :: fileobj !< File object
   integer, intent(in), optional :: unlim_dim_level !< Unlimited dimension
                                                      !! level.
-  integer :: i !< No description
+  integer :: i, unit !< No description
```
Comment (climbfuji) [inline fms2_io/netcdf_io.F90]: If unit is not needed for other compilers, it better goes into an `ifdef` section as well.

**Review by climbfuji — COMMENTED** [review]
I don't think we'll be able to get this change into the official FMS (simply deactivating certain parts of the code). Do you know what the proper fix would be?

File: fms2_io/netcdf_io.F90 (hunk 1 above)
Comment (benjaminmenetrier) [reply fms2_io/netcdf_io.F90]: Corrected.

**Comment by symoore90** [comment]
What is the status of this issue?

**Comment by climbfuji** [comment]
The target branch for this PR will soon be dropped from JEDI as we move to the official FMS releases from NOAA-GFDL. The codebase is different enough that it makes sense to close this PR. Besides, the nvidia compilers are not a priority for us at the moment.

**Pull request closed without merging**